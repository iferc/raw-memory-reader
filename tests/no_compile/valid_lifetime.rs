use raw_memory_ref::RawMemoryRef;

fn main() {
    let value = vec![5i32];
    let boxed =
        unsafe { RawMemoryRef::new(&value).into_inner_with_length(std::mem::size_of::<i32>()) };

    // should be fine here
    assert_eq!(boxed.allocated_bytes(), vec![5u8, 0, 0, 0]);

    // original value dropped
    drop(value);

    // cannot compil due to read after drop
    assert_eq!(boxed.allocated_bytes(), vec![5u8, 0, 0, 0]);
}
