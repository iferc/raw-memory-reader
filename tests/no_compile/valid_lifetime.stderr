error[E0505]: cannot move out of `value` because it is borrowed
  --> tests/no_compile/valid_lifetime.rs:12:10
   |
6  |         unsafe { RawMemoryRef::new(&value).into_inner_with_length(std::mem::size_of::<i32>()) };
   |                                    ------ borrow of `value` occurs here
...
12 |     drop(value);
   |          ^^^^^ move out of `value` occurs here
...
15 |     assert_eq!(boxed.allocated_bytes(), vec![5u8, 0, 0, 0]);
   |                ----------------------- borrow later used here
